### ISSUE TITLE
Add function call support

### ISSUE THREAD

**Issue by yaroslavyaroslav** [yaroslavyaroslav/OpenAI-sublime-text#17]
OpenAI added support for a functions calling. I believe in a big potential of this feature related to a sublime text, since its Python api are hittable with such text pipeline.

Anyway this would be a long term task, in addition to the lack of time in nearest future, this thing worth to take some time and effort to design it appropriately.

Related to [this](https://platform.openai.com/docs/guides/gpt/function-calling).

**Comment by yaroslavyaroslav** [comment]
Was implemented in #92

**Comment by yaroslavyaroslav** [comment]
So just to keep y'all updated. I have implemented this thing like a few month ago, and it works pretty well actually. 

However to make it a little bit more reliable I've decided to move to types rather than to keep working with python Dict within the code. And it appears that it's unbearable to do so reliably in Python. Like the last drop that overflow my potion was the constant occurrence of Dict without any error when type hint have had forced a concrete type. 

So I gave up and reimplemented the core in rust, which in addition to that allows me to use the god damn dependencies. Obviously it took time, but the task is almost over and I expect it to be released pretty soon.
